internal/consensus/common_test.go

```diff
@@ -190,22 +189,36 @@ func incrementRound(vss ...*validatorStub) {
 	}
 }
 
-func sortVValidatorStubsByPower(ctx context.Context, vss []*validatorStub) {
-	sort.Slice(vss, func(i, j int) bool {
-		vssi, err := vss[i].GetPubKey(ctx)
+type ValidatorStubsByPower []*validatorStub
```

Comment:
If you want, you can still inject a context:

```go
type byPower struct{
   ctx  context.Context
   vals []*Whatever
}

func (b byPower) Len() int { return len(b.vals) }
func (b byPower) Swap(i, j int) {
    // do stuff with b.ctx and b.vals to swap
}
// etc.
```

Reply:
that's fair, I think my method also works, and avoids needing to put a context in a struct.